firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Homepage Rebuild/HomepageViewControllerTests.swift

```diff
@@ -267,11 +320,39 @@ final class HomepageViewControllerTests: XCTestCase, StoreTestUtility {
         )
 
         // Need to call loadViewIfNeeded and newState to populate the datasource
-        // layoutIfNeeded() recalculates the collection view to have items
         subject.loadViewIfNeeded()
-        subject.newState(state: initialState)
-        subject.view.layoutIfNeeded()
+        subject.newState(state: newState)
+
+        XCTAssertTrue(newState.shouldTriggerImpression)
+        XCTAssertTrue(mockThrottler.didCallThrottle)
+        let actionCalled = try XCTUnwrap(
+            mockStore.dispatchedActions.last(where: { $0 is HomepageAction }) as? HomepageAction
+        )
+        let actionType = try XCTUnwrap(actionCalled.actionType as? HomepageActionType)
+        XCTAssertEqual(actionType, HomepageActionType.sectionSeen)
+        XCTAssertEqual(actionCalled.windowUUID, .XCTestDefaultUUID)
+    }
+
+    // This test differs from the one above in that is has the `stories-redesign` feature flag enabled.
+    // When that flag is enabled, we need to add some visible sections in the collection view to trigger impression
+    // telemetry, whereas the test above can rely on the, always visible, customize homepage button section to be present.
+    func test_newState_forTriggeringImpression_withStoriesRedesignEnabled_triggersHomepageAction() async throws {
```

Comment:
Would be nice to also see a case where we don't trigger the call when we don't populate a section state.